### PR TITLE
Add integration test cases for mapcidr

### LINKED ISSUE #109: Implement test cases
## Description
The repository should be covered by integration tests covering various option usages

### PR DESCRIPTION
**Pull request by ShubhamRasal**
Closes #109

### CODE REVIEW

File: cmd/integration-test/mapcidr.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"IPRange & CIDR Aggregation(file)":   &mapCidrQuery{question: "", expectedOutput: []string{"166.8.0.0/14"}, args: "-cl ./goldenfiles/ip_cidr.txt -a"},
+	"IPRange & CIDR Expansion":           &mapCidrQuery{question: "192.168.0.0/30,192.168.0.4-192.168.0.10", expectedOutput: []string{"192.168.0.0", "192.168.0.1", "192.168.0.2", "192.168.0.3", "192.168.0.4", "192.168.0.5", "192.168.0.6", "192.168.0.7", "192.168.0.8", "192.168.0.9", "192.168.0.10"}},
+	"IPRange & CIDR Count":               &mapCidrQuery{question: "166.8.0.0/16,166.11.0.0/16,166.9.0.0-166.10.255.255", expectedOutput: []string{"262144"}, args: "-c"},
+	"IPRange & CIDR convert IPs to IPv6": &mapCidrQuery{question: "192.168.0.0-192.168.0.3", expectedOutput: []string{"00:00:00:00:00:ffff:c0a8:0000", "00:00:00:00:00:ffff:c0a8:0001", "00:00:00:00:00:ffff:c0a8:0002", "00:00:00:00:00:ffff:c0a8:0003"}, args: "-t6"},
+
+	"OutputFile case": &mapCidrQueryOutputFile{question: "192.168.0.0/30", expectedOutput: []string{"192.168.0.0", "192.168.0.1", "192.168.0.2", "192.168.0.3"}, args: "-o /tmp/output.txt", outputfile: "/tmp/output.txt"},
+}
+
+func (h *mapCidrQuery) Execute() error {
+
```
Comment (Mzack9999) [inline cmd/integration-test/mapcidr.go:86]: Would it be possible to remove the empty line?

File: cmd/integration-test/mapcidr.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	result, err := RunMapCidrAndGetResults(h.question, debug, h.args)
+	if err != nil {
+		return err
+	}
+	err = compareResult(h.expectedOutput, result)
```
Comment (Mzack9999) [inline cmd/integration-test/mapcidr.go:91]:
```go
return compareResult(h.expectedOutput, result)
```

File: cmd/integration-test/mapcidr.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *mapCidrQueryOutputFile) Execute() error {
+
```
Comment (Mzack9999) [inline cmd/integration-test/mapcidr.go:99]: Would it be possible to remove the empty line?

File: cmd/integration-test/mapcidr.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		return err
+	}
+	result := []string{}
+	items := strings.Split(string(fileContent), "\n")
+	for _, i := range items {
+		if i != "" {
+			result = append(result, i)
+		}
+	}
+	err = compareResult(h.expectedOutput, result)
```
Comment (Mzack9999) [inline cmd/integration-test/mapcidr.go:116]:
```go
return compareResult(h.expectedOutput, result)
```

File: cmd/integration-test/mapcidr.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+func compareResult(expected, result []string) error {
+	// check if incorrent number of result
+	if len(result) != len(expected) {
+		return errIncorrectResultsCount(result)
+	}
+	var notEqual = false
+	for _, v := range result {
+		if !slices.Contains(expected, v) {
+			notEqual = true
```
Comment (Mzack9999) [inline cmd/integration-test/mapcidr.go:139]: I think we can return directly here:
```go
return errIncorrectResult(expected, result)
```

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
lgtm - suggesting minor flow optimizations

File: cmd/integration-test/mapcidr.go (hunk 1 above)
Comment (ShubhamRasal) [reply cmd/integration-test/mapcidr.go]: Removed line as per your suggestion.

File: cmd/integration-test/mapcidr.go (hunk 3 above)
Comment (ShubhamRasal) [reply cmd/integration-test/mapcidr.go]: Removed line as per your suggestion.

File: cmd/integration-test/mapcidr.go (hunk 2 above)
Comment (ShubhamRasal) [reply cmd/integration-test/mapcidr.go]: updated the code as per your suggestion.

File: cmd/integration-test/mapcidr.go (hunk 4 above)
Comment (ShubhamRasal) [reply cmd/integration-test/mapcidr.go]: updated the code as per your suggestion.

**Review by Mzack9999 — APPROVED** [review]
lgtm!

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**